js/src/admin/components/AdminPage.tsx

```diff
@@ -0,0 +1,264 @@
+import type Mithril from 'mithril';
+
+import app from '../app';
+import Page from '../../common/components/Page';
+import Button from '../../common/components/Button';
+import Switch from '../../common/components/Switch';
+import Select from '../../common/components/Select';
+import classList from '../../common/utils/classList';
+import Stream from '../../common/utils/Stream';
+import saveSettings from '../utils/saveSettings';
+import AdminHeader from './AdminHeader';
+
+export interface AdminHeaderOptions {
+  title: string;
+  description: string;
+  icon: string;
+  /**
+   * Will be used as the class for the AdminPage.
+   *
+   * Will also be appended with `-header` and set as the class for the `AdminHeader` component.
+   */
+  className: string;
+}
+
+interface CommonSettingsItemOptions extends Mithril.Attributes {
+  setting: string;
+  label: string | ReturnType<typeof app.translator.trans>;
```

Comment:
Not sure how I feel about explicitly stating that it can be output of trans method... I mean that's not wrong but I don't think that's how we should refer to it 🤔 

Label also can be any mithril child... I think we should allow for that (whatever the type is specifically, I always forget)
Same with help perhaps?